### DISCUSSION TITLE
simple "getting started" questions

### DISCUSSION THREAD

**Discussion by barnett3** [piccolo-orm/piccolo#168] [Q&A]
Hi,

I'm relatively new to python but have used sql for years. I'd like to try piccolo by writing an interface for my old music database (currently in sqlite, but I'm happy to try postgres). I couldn't tell from the docs:
1. when I put the schema in tables.py, is piccolo supposed to _create_ the tables & columns or do I need to do that in pg with create table, etc.?
2. when I move my existing data from sqlite to pg, am I supposed to let piccolo manage the foreign keys or can I use my existing keys (which wouldn't be serial types)?
3. is the migration tool supposed to help with this?
4. Is there another place besides Read the Docs to look for info?

Thank you!
Barnett

**Comment by sinisaos** [comment]
@barnett3 Maybe you can try this #88.

**Comment by aminalaee** [comment]
@barnett3 For question No. 2, you can define your own primary keys, there's experimental support for custom primary key definitions (other than `Serial`).
https://piccolo-orm.readthedocs.io/en/latest/piccolo/schema/defining.html#primary-key

**Reply by barnett3** [reply]
thanks for these answers! So it sounds like I can try to autogenerate a schema (or just type it out) and potentially use my existing sqlite file. But am I correct that the way it's supposed to work is that I create the schema, and an empty postgres db myself - and then use piccolo to populate the data into the db?

**Reply by sinisaos** [reply]
@barnett3 I haven't tried this, but this maybe can work. First migrate sqlite to postgres with any of [this ways](https://www.google.com/search?q=convert+sqlite+to+postgres&oq=convert+sqlite+to+postgres&aqs=chrome..69i57.9328j0j1&sourceid=chrome&ie=UTF-8) and second try #88. I hope this helps you.
